serial/serialwin32.py

```diff
@@ -316,9 +325,22 @@ def write(self, data):
                 if not success and win32.GetLastError() not in (win32.ERROR_SUCCESS, win32.ERROR_IO_PENDING):
                     raise SerialException("WriteFile failed ({!r})".format(ctypes.WinError()))
 
-                # Wait for the write to complete.
-                #~ win32.WaitForSingleObject(self._overlapped_write.hEvent, win32.INFINITE)
-                win32.GetOverlappedResult(self._port_handle, self._overlapped_write, ctypes.byref(n), True)
+                timeout = Timeout(self._timeout)
+                while True:
+                    # even with timeout infinite, return from kernel mode every 100ms to allow for things like KeyboardInterrupt to happen
+                    millis = win32.DWORD(int(min(timeout.time_left() or 0.1, 0.1) * 1000))
```

Comment:
if time_left() is 0, millis will be 0
if time_left() is < 0.1, millis will be time_left() * 1000
if time_left() is > 0.1, millis will be 0.1 * 1000 (capped)